meetings/working-groups/collection-literals/CL-LDM-2023-08-14.md

```diff
@@ -0,0 +1,205 @@
+static void MyCollectionAndArray<T>(T[] args) { }
+```
+
+A single rule is added to [*better conversion from expression*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#11644-better-conversion-from-expression) to handle the preference order above: see [proposal](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#overload-resolution).
+
+The added rule applies to *collection expression arguments only*.
+* The rule is not a breaking change to existing code.
+* Changing grouping is a *breaking change*.
+* Adding target types is a *breaking change*.
+
+**Question**: Should the first group include all `ref struct` types, not just spans?
+
+**Question**: Should we differentiate between arrays and concrete types? If so, how?
+
+*An array or concrete type is already preferred over an interface if the array or concrete type implements that interface. And if the array or concrete type does not implement the interface, we shouldn't prefer one over the other. Instead, the overload resolution proposal should be: Prefer a span over an array or interface if the iteration type of the span is implicitly convertible to the iteration type of the array or interface.*
+
+## Type inference
+
+*See proposal: [Type inference](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#type-inference)*
+
+Method type inference involving collection expressions allows inferring the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of a collection type. There is no support for inferring the containing *collection type*.
+
+The *iteration type* is defined by `foreach`. If a *collection initializer type* implements `IEnumerable` only, the iteration type is `object`.
+
+Inferences are made to the iteration type, independently for each element. This is analogous to element inference for *tuples*.
+
+```c#
+AsArray([null]);             // error: cannot infer T
+AsArray([1, 2, 3]);          // AsArray<int>(int[])
+AsListOfArray([[4, 5], []]); // AsListOfArray<int>(List<int[]>)
+
+static T[] AsArray<T>(T[] arg) => arg;
+static List<T[]> AsListOfArray<T>(List<T[]> arg) => arg;
+```
+
+The [*type inference*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1163-type-inference) change is handled with a rule for *input type inference* and a corresponding rule for *output type inference*: see [proposal](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#type-inference).
+
+*Input type inference* is necessary for walking into the element expressions to infer from explicitly-typed lambda parameters, from tuple element types, or from nested collection expressions.
+```c#
+InputTypeInference([(int y) => { }]); // InputTypeInference<int>()
+
+static void InputTypeInference<T>(List<Action<T>> x) { }
```

Comment:
woudl this work with a different delegate tyep from Func/Action?

Reply:
This doesn't depend on `Action<T>`. A delegate such as `delegate void D<T>(T t);` would work too.